clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/MarathonPathId.groovy

```diff
@@ -0,0 +1,140 @@
+package com.netflix.spinnaker.clouddriver.dcos.deploy.util.id
+
+import com.google.common.base.Strings
+
+import static com.google.common.base.Preconditions.checkArgument
+import static com.google.common.base.Strings.nullToEmpty
+import static java.util.stream.Collectors.joining
+
+/**
+ * Represents a hierarchical Marathon application identifier.
+ */
+class MarathonPathId {
+    public final static String PART_PATTERN = $/[a-z0-9]+(-+[a-z0-9]+)*/$
```

Comment:
I went back and forth with this one, but ultimately decided to not include them since we weren't going to consider them valid within context of Spinnaker. Felt it best given this isn't a marathon library but just a helper within Spinnaker, to just run with what we would allow within Spinnaker (something we DO need to document in our naming document thingy though).